pennylane_qiskit/vqe/vqe_runner.py

```diff
@@ -0,0 +1,548 @@
+
+        num_qubits = num_qubits_h
+
+        ansatz_circ = getattr(lib_local, ansatz, None)
+        if not ansatz_circ:
+            raise ValueError(f"Ansatz {ansatz} not in n_local circuit library.")
+
+        inputs["ansatz"] = ansatz
+        inputs["ansatz_config"] = ansatz_config
+
+        # If given x0, validate its length against num_params in ansatz:
+        x0 = np.asarray(x0)
+        ansatz_circ = ansatz_circ(num_qubits, **ansatz_config)
+        num_params = ansatz_circ.num_parameters
+
+        if x0.shape[0] != num_params:
+            x0 = 2 * np.pi * np.random.rand(num_params)
+
+        inputs["x0"] = x0
+
+    # Transform the PennyLane hamilonian to a suitable form
+    hamiltonian = hamiltonian_to_list_string(hamiltonian, num_qubits)
+
+    inputs["hamiltonian"] = hamiltonian
+
+    # Set the rest of the inputs
+    inputs["optimizer"] = optimizer
+    inputs["optimizer_config"] = optimizer_config
+    inputs["shots"] = shots
+    inputs["use_measurement_mitigation"] = use_measurement_mitigation
+
+    # Specify a single hub, group and project
+    hub = kwargs.get("hub", "ibm-q")
+    group = kwargs.get("group", "open")
+    project = kwargs.get("project", "main")
+
+    provider = IBMQ.get_provider(hub=hub, group=group, project=project)
+
+    rt_job = RuntimeJobWrapper()
+
+    # Callbacks functions are different between optimizers.
+    if optimizer in ["SPSA", "QNSPSA"]:
+        job = provider.runtime.run(
+            program_id, options=options, inputs=inputs, callback=rt_job._callback
+        )
+    else:
+        job = provider.runtime.run(
+            program_id, options=options, inputs=inputs, callback=rt_job._scipycallback
+        )
+    rt_job._job = job
+
+    return rt_job
+
+
+def hamiltonian_to_list_string(hamiltonian, num_qubits):
+    r"""Convert a hamiltonian from PennyLane to a list of coefficient and strings.
+
+    Args:
+        hamiltonian (qml.Hamiltonian): An Hamiltonian from PennyLane.
+        num_qubits (int): Number of qubits.
+
+    Returns:
+        list[tuple[float,str]]: Hamiltonian in a format for the runtime program.
+    """
+
+    coeff, observables = hamiltonian.terms
+
+    authorized_obs = {"PauliX", "PauliY", "PauliZ", "Hadamard", "Identity"}
+
+    for obs in observables:
+        if isinstance(obs.name, list):
+            for ob in obs.name:
+                if ob not in authorized_obs:
+                    raise qml.QuantumFunctionError("Obs not accepted")
+        else:
+            if obs.name not in authorized_obs:
+                raise qml.QuantumFunctionError("Obs not accepted")
+
+    # Create string Hamiltonian
+    obs_str = {"PauliX": "X", "PauliY": "Y", "PauliZ": "Z", "Hadamard": "H", "Identity": "I"}
+
+    obs_org = []
+    for obs in observables:
+        if isinstance(obs.name, list):
+            internal = []
+            for i, j in zip(obs.wires.tolist(), obs.name):
+                internal.append([i, obs_str[j]])
+            internal.sort()
```

Comment:
Why do we sort?

Reply:
For example: PauliY(1) @ PauliY(0) -> [[[0,'Y'], [1,'Y']]]